### ISSUE TITLE
Make choco get its proxy settings also from environment Variables

### ISSUE THREAD

**Issue by c33s** [chocolatey/choco#605]
as referenced in the issue #553 you linked the manual for proxy setup https://github.com/chocolatey/choco/wiki/Proxy-Settings-for-Chocolatey

it would be very good, for a automation tool, to get the proxy settings from the environmental variables:

```
HTTPS_PROXY=http://proxy.example.com:1080
HTTP_PROXY=http://proxy.example.com:1080
```

**Comment by ferventcoder** [comment]
you mean use those if they are present?

**Comment by c33s** [comment]
yes, exactly

**Comment by majkinetor** [comment]
I second this. This is important for x-platform compatibility too. Linux folks would appreciate this.

Just as a reference, here is my [Update-CLIProxy](https://github.com/majkinetor/posh/blob/master/MM_Network/Update-CLIProxy.ps1)  that creates those variables from explorer settings. I have it in my $PROFILE so linux tools (gem, bundler, pip etc.) work.

**Comment by abmoham** [comment]
I'd like to request that it also respect proxy bypass:

NO_PROXY=127.0.0.1,localhost,*.example.com

**Comment by ferventcoder** [comment]
This will be implemented in 0.10.4
